src/Form/LinkBlockFormDataProvider.php

```diff
@@ -104,7 +106,16 @@ public function __construct(
     public function getData()
     {
         if (null === $this->idLinkBlock) {
-            return [];
+            $shopSelectedByDefault = [$this->shopContext->getContextShopID()];
+            if (!$this->shopContext->isSingleShopContext()) {
+                $shopSelectedByDefault = $this->shopContext->getContextListShopID();
+            }
```

Comment:
```suggestion
            if (!$this->shopContext->isSingleShopContext()) {
                $shopSelectedByDefault = $this->shopContext->getContextListShopID();
            } else {
                $shopSelectedByDefault = [$this->shopContext->getContextShopID()];
            }
```

To avoid execute SQL requests for nothing